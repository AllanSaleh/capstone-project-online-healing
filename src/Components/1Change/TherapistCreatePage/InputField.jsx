import React from 'react';

const InputField = ({ id, title, inputType, placeholder }) => (
  <div className="my-4">
    <label htmlFor={id} className="text-subtitle capitalize text-black text-opacity-50">
      {title}
    </label>
    <br />
    <input
      className="max-h-8 py-6 px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
      placeholder={placeholder}
      type={inputType}
      id={id}
      name={id}
      required
    />
  </div>
);

export default InputField;
