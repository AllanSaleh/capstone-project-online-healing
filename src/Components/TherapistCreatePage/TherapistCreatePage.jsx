import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import InputField from './InputField';

export default function TherapistCreatePage() {
  window.scrollTo(0, 0);
  const history = useHistory();
  const { t } = useTranslation();

  const GoToThankYou = () => {
    // Code to validate inputs

    // Firebase stuff to send new therapist request

    history.push({
      pathname: '/ThankYou',
      state: [t('Create.Thanks1'), t('Create.Thanks2')],
    });
  };

  return (
    <div className="px-sides pt-navbar">
      <h1 className="text-xl md:text-title md:py-4 uppercase">{t('Create.title')}</h1>
      <form className="flex flex-col md:w-10/12 md:mx-auto lg:w-2/5 lg:m-0">
        <InputField id="user-name" title="user name" inputType="text" placeholder="Exp: John" />
        <InputField
          id="email"
          title="email"
          inputType="email"
          placeholder="Exp: someone@something.com"
        />
        <InputField id="city" title="city" inputType="text" placeholder="Exp: Erbil" />
        <InputField
          id="license"
          title="license number"
          inputType="text"
          placeholder="Exp: 123456789"
        />
        <InputField id="password" title="password" inputType="password" placeholder="Password" />
        <InputField
          id="confirm-password"
          title="confirm password"
          inputType="password"
          placeholder="re-type your password"
        />
        <button
          onClick={() => GoToThankYou()}
          type="button"
          className="text-subtitle bg-blue-dark px-4 py-2 my-16 w-full md:w-48 md:mx-auto border rounded-lg uppercase hover:bg-transparent hover:border-blue-dark hover:text-blue-dark hover:shadow-lg transition-all duration-300"
        >
          {t('Create.Create')}
        </button>
      </form>
    </div>
  );
}
