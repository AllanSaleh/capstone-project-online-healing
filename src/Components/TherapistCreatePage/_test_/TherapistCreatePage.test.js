import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TherapistCreatePage from '../TherapistCreatePage';

it('Therapist Create Page renders correctly', () => {
  const tree = renderer.create(<TherapistCreatePage />).toJSON();
  expect(tree).toMatchSnapshot();
});
