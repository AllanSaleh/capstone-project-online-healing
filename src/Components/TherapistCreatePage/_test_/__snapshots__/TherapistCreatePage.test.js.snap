// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-sides pt-navbar"
>
  <h1
    className="text-xl md:text-title md:py-4 uppercase"
  >
    create an account
  </h1>
  <form
    className="flex flex-col md:w-10/12 md:mx-auto lg:w-2/5 lg:m-0"
  >
    <div
      className="my-4"
    >
      <label
        className="text-md md:text-subtitle capitalize text-black text-opacity-50"
        htmlFor="user-name"
      >
        user name
      </label>
      <br />
      <input
        className="text-sm md:text-md lg:text-xl max-h-8 py-3 px-2 md:py-6 md:px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
        id="user-name"
        name="user-name"
        placeholder="Exp: John"
        required={true}
        type="text"
      />
    </div>
    <div
      className="my-4"
    >
      <label
        className="text-md md:text-subtitle capitalize text-black text-opacity-50"
        htmlFor="email"
      >
        email
      </label>
      <br />
      <input
        className="text-sm md:text-md lg:text-xl max-h-8 py-3 px-2 md:py-6 md:px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
        id="email"
        name="email"
        placeholder="Exp: someone@something.com"
        required={true}
        type="email"
      />
    </div>
    <div
      className="my-4"
    >
      <label
        className="text-md md:text-subtitle capitalize text-black text-opacity-50"
        htmlFor="city"
      >
        city
      </label>
      <br />
      <input
        className="text-sm md:text-md lg:text-xl max-h-8 py-3 px-2 md:py-6 md:px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
        id="city"
        name="city"
        placeholder="Exp: Erbil"
        required={true}
        type="text"
      />
    </div>
    <div
      className="my-4"
    >
      <label
        className="text-md md:text-subtitle capitalize text-black text-opacity-50"
        htmlFor="license"
      >
        license number
      </label>
      <br />
      <input
        className="text-sm md:text-md lg:text-xl max-h-8 py-3 px-2 md:py-6 md:px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
        id="license"
        name="license"
        placeholder="Exp: 123456789"
        required={true}
        type="text"
      />
    </div>
    <div
      className="my-4"
    >
      <label
        className="text-md md:text-subtitle capitalize text-black text-opacity-50"
        htmlFor="password"
      >
        password
      </label>
      <br />
      <input
        className="text-sm md:text-md lg:text-xl max-h-8 py-3 px-2 md:py-6 md:px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
        id="password"
        name="password"
        placeholder="type your password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="my-4"
    >
      <label
        className="text-md md:text-subtitle capitalize text-black text-opacity-50"
        htmlFor="confirm-password"
      >
        confirm password
      </label>
      <br />
      <input
        className="text-sm md:text-md lg:text-xl max-h-8 py-3 px-2 md:py-6 md:px-4 mt-2 w-full border border-black border-opacity-20 rounded-lg outline-none focus:border-blue-dark transition duration-200"
        id="confirm-password"
        name="confirm-password"
        placeholder="re-type your password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="text-subtitle bg-blue-dark px-4 py-2 my-16 w-full md:w-48 md:mx-auto border rounded-lg uppercase hover:bg-transparent hover:border-blue-dark hover:text-blue-dark hover:shadow-lg transition-all duration-300"
      type="button"
    >
      create
    </button>
  </form>
</div>
`;
